application/common/components/Emailer.php

```diff
@@ -776,21 +776,38 @@ public function sendPasswordExpiringEmails()
             'status' => 'starting',
         ];
 
-        $users = User::getActiveUnlockedUsers();
+        $users = \Yii::$app->getDb()->createCommand("SELECT u.*
+            FROM `user` u
+                JOIN `password` p ON u.id = p.user_id
+                LEFT JOIN `email_log` e ON u.id = e.user_id
+                    AND e.message_type = 'password-expiring'
+                    AND e.sent_utc >= CURRENT_DATE() - INTERVAL ? DAY
+            WHERE u.active = 'yes'
+                AND u.locked = 'no'
+                AND p.expires_on < CURRENT_DATE() + INTERVAL 15 DAY
+                AND p.expires_on >= CURRENT_DATE() # send a different message if expired
+                AND e.id IS NULL
+            GROUP BY u.id
+            HAVING COUNT(*) = 1;")
+            ->bindValue(1, $this->emailRepeatDelayDays)
```

Comment:
Is there an 'emailRepeatDelayDays' field in the user table? Does this create a calculated field?

Reply:
Oh wait. That "INTERVAL ? DAY" is what is being bound?

Reply:
Yep